src/Microsoft.AspNetCore.Mvc.Razor/Internal/FileProviderRazorProject.cs

```diff
@@ -58,26 +50,23 @@ private IEnumerable<RazorProjectItem> EnumerateFiles(IDirectoryContents director
         {
             if (directory.Exists)
             {
-                foreach (var file in directory)
+                foreach (var fileInfo in directory)
                 {
-                    if (file.IsDirectory)
+                    if (fileInfo.IsDirectory)
                     {
-                        var relativePath = prefix + "/" + file.Name;
+                        var relativePath = prefix + "/" + fileInfo.Name;
                         var subDirectory = _provider.GetDirectoryContents(JoinPath(basePath, relativePath));
                         var children = EnumerateFiles(subDirectory, basePath, relativePath);
                         foreach (var child in children)
                         {
                             yield return child;
                         }
                     }
-                    else if (string.Equals(RazorFileExtension, Path.GetExtension(file.Name), StringComparison.OrdinalIgnoreCase))
+                    else if (string.Equals(RazorFileExtension, Path.GetExtension(fileInfo.Name), StringComparison.OrdinalIgnoreCase))
                     {
-                        var filePath = prefix + "/" + file.Name;
-                        var absoluteBasePath = _hostingEnvironment.ContentRootPath;
-                        var relativePhysicalPath = file.PhysicalPath?.Substring(absoluteBasePath.Length + 1); // Include leading separator
-                        relativePhysicalPath = relativePhysicalPath ?? filePath; // Use the incoming path if the file is not directly accessible
+                        var filePath = prefix + "/" + fileInfo.Name;
```

Comment:
Please use Path.Combine instead

Reply:
We want this path to always have forward slashes. Path.Combine would normalize to OS specific separator. We do that for PhysicalPath